syntax = "proto3";

package com.project.scheduleService;

import "google/protobuf/timestamp.proto";

message ScheduleRequest {
  int64 groupId = 1;
}

message ScheduleResponse {
  int64 id = 1;
  string title = 2;
  string teacher = 3;
  string auditorium = 4;
  google.protobuf.Timestamp timeStart = 5;
  google.protobuf.Timestamp timeEnd = 6;
  string weekType = 7;
  string dayType = 8;
  int64 groupId = 9;
}

service ScheduleService {
  rpc getScheduleTomorrow(ScheduleRequest) returns (stream ScheduleResponse);
  rpc getScheduleToday(ScheduleRequest) returns (stream ScheduleResponse);
  rpc getScheduleWeek(ScheduleRequest) returns (stream ScheduleResponse);
}